frame/lending/src/tests.rs

```diff
@@ -1,50 +1,55 @@
-#![allow(unused_imports)]
-
-//! Test for Lending. Runtime is almost real.
-//! TODO: cover testing events - so make sure that each even is handled at least once
 //! (events can be obtained from System pallet as in banchmarking.rs before this commit)
 //! TODO: OCW of liquidations (like in Oracle)
 //! TODO: test on small numbers via proptests - detect edge case what is minimal amounts it starts
 //! to accure(and miminal block delta), and maximal amounts when it overflows
 
-use composable_traits::lending::{Lending as LendingTrait, RepayStrategy, TotalDebtWithInterest};
-use frame_benchmarking::Zero;
-use std::ops::{Div, Mul};
-
 use crate::{
 	self as pallet_lending, accrue_interest_internal, currency::*, mocks::*,
 	models::borrower_data::BorrowerData, setup::assert_last_event, AccruedInterest, Error,
 	MarketIndex,
 };
-use codec::{Decode, Encode};
-use composable_support::validation::{TryIntoValidated, Validated};
-use composable_tests_helpers::{prop_assert_acceptable_computation_error, prop_assert_ok};
+use composable_support::validation::TryIntoValidated;
+use composable_tests_helpers::{prop_assert_acceptable_computation_error, prop_assert_ok, test};
 use composable_traits::{
-	defi::{CurrencyPair, LiftedFixedBalance, MoreThanOneFixedU128, Rate, ZeroToOneFixedU128},
-	lending::{self, math::*, CreateInput, UpdateInput, UpdateInputVaild},
+	defi::{
+		CurrencyPair, DeFiComposableConfig, LiftedFixedBalance, MoreThanOneFixedU128, Rate,
+		ZeroToOneFixedU128,
+	},
+	lending::{
+		math::*, CreateInput, Lending as LendingTrait, RepayStrategy, TotalDebtWithInterest,
+		UpdateInput, UpdateInputVaild,
+	},
 	oracle,
 	time::SECONDS_PER_YEAR_NAIVE,
 	vault::{self, Deposit, VaultConfig},
 };
 use frame_support::{
 	assert_err, assert_noop, assert_ok,
 	dispatch::{DispatchErrorWithPostInfo, DispatchResultWithPostInfo, PostDispatchInfo},
-	traits::fungibles::{Inspect, Mutate},
+	traits::{
+		fungibles::{Inspect, Mutate},
+		OriginTrait,
+	},
 	weights::Pays,
 };
 use frame_system::{EventRecord, Phase};
 use pallet_vault::models::VaultInfo;
 use proptest::{prelude::*, test_runner::TestRunner};
 use sp_arithmetic::assert_eq_error_rate;
-use sp_core::{H256, U256};
+use sp_core::U256;
 use sp_runtime::{
 	ArithmeticError, DispatchError, FixedPointNumber, FixedU128, ModuleError, Percent, Perquintill,
 };
+use std::ops::{Div, Mul};
 
 const DEFAULT_MARKET_VAULT_RESERVE: Perquintill = Perquintill::from_percent(10);
 const DEFAULT_MARKET_VAULT_STRATEGY_SHARE: Perquintill = Perquintill::from_percent(90);
 const DEFAULT_COLLATERAL_FACTOR: u128 = 2;
 
+type SystemAccountId<R> = <R as frame_system::Config>::AccountId;
+type SystemOrigin<R> = <R as frame_system::Config>::Origin;
+type SystemEvent<R> = <R as frame_system::Config>::Event;
```

Comment:
unofficial naming convention for these kinds of type definitions is `TypeNameOf<T>`, so `SystemAccountIdOf<T>`

Reply:
Done